contrib/postgres_fdw/postgres_fdw.c

```diff
@@ -1587,6 +1597,63 @@ postgresBeginForeignScan(ForeignScanState *node, int eflags)
 	fsstate->async_capable = node->ss.ps.async_capable;
 }
 
+void *
+postgresGetForeignObject(const char *schemaname, const char *relname, Oid srvowner, Oid srvoid, int *numattr)
+{
+	UserMapping *user;
+	PGconn	   *fsconn;
+	PgFdwScanState *fsstate;
+	char		*query;
+	PGresult	*last_result;
+
+	/*
+	 * We'll save private state in node->fdw_state.
+	 */
+	fsstate = (PgFdwScanState *) palloc0(sizeof(PgFdwScanState));
+
+	/* Get info about foreign table. */
+	user = GetUserMapping(srvowner, srvoid);
+
+	/*
+	 * Get connection to the foreign server.  Connection manager will
+	 * establish new connection if necessary.
+	 */
+	fsconn = GetConnection(user, false, &fsstate->conn_state);
+
+	/* Assign a unique ID for my cursor */
+	fsstate->cursor_number = GetCursorNumber(fsstate->conn);
+	fsstate->cursor_exists = false;
+
+	query = palloc(strlen("SELECT * FROM .") + strlen(schemaname) + strlen(relname) + 10);
+	sprintf(query, "SELECT * FROM %s.%s", schemaname, relname);
+	fsstate->query = query;
+
+	last_result = PQexec(fsconn, fsstate->query);
+
+	*numattr = PQnfields(last_result);
+
+	pfree(fsstate);
+	ReleaseConnection(fsconn);
+
+	return (void *)last_result;
```

Comment:
first, PQresult need  to clear by PQclear（）。 but， I  can't find PQclear anywhere.
second,  PQconnection need to close by PQfinish() also.